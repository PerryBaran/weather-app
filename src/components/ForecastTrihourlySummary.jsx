import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";
import "../styles/ForecastTrihourlySummary.css";
import useUnits from "../helpers/getUnits";
import useUnitContext from "../hooks/useUnitContext";

function ForecastTrihourlySummary({ time, weather, icon, temp }) {
  const { units } = useUnitContext();
  const { tempUnits } = useUnits(units);
  const description = weather ? weather.split(" ") : [weather];

  return (
    <div
      className="forecast-trihourly-summary"
      data-testid="forecast-trihourly-summary"
    >
      <p className="forecast-trihourly-summary__time">{time}</p>
      {!icon ? (
        <p className="forecast-trihourly-summary__no-data">
          No <br />
          Data
        </p>
      ) : (
        <>
          <div
            className="forecast-trihourly-summary__icon"
            data-testid="forecast-icon"
          >
            <WeatherIcon name="owm" iconId={icon} />{" "}
          </div>
          <div className="forecast-trihourly-summary__description">
            {description.length === 1 ? (
              <p>{description[0]}</p>
            ) : (
              <p>
                {description[0]} <br />
                {description[1]}
              </p>
            )}
          </div>
          <p className="forecast-trihourly-summary__temp">
            {temp.toFixed(1)}
            {tempUnits}
          </p>
        </>
      )}
    </div>
  );
}

ForecastTrihourlySummary.defaultProps = {
  icon: null,
  temp: null,
  weather: null,
};

ForecastTrihourlySummary.propTypes = {
  time: PropTypes.string.isRequired,
  icon: PropTypes.string,
  temp: PropTypes.number,
  weather: PropTypes.string,
};

export default ForecastTrihourlySummary;
