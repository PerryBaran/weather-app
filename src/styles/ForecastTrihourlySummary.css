.forecast-trihourly-summary {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-items: center;
  align-items: center;
  gap: 10px;
}

.forecast-trihourly-summary:first-child {
  border-left: none;
}

.forecast-trihourly-summary__time {
  font-size: 1rem;
}

.forecast-trihourly-summary__icon {
  font-size: min(4rem, 5vw);
}

.forecast-daily-summary__description {
  font-size: min(1rem, 2.5vw);
}

.forecast-trihourly-summary__temp {
  font-size: min(1rem, 2.5vw);
}


@media screen and (max-width: 780px) {
  .forecast-trihourly-summary__date {
    width: 50px;
  }
}