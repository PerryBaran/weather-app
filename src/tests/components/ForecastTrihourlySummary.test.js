import React from "react";
import { render } from "@testing-library/react";
import ForecastTrihourlySummary from "../../components/ForecastTrihourlySummary";

xdescribe("ForecastTrihourlySummary", () => {
  const validProps = {
    time: "now",
    icon: "200",
    temp: 34,
    weather: "hot",
  };

  describe("snapshot", () => {
    it("renders corretly", () => {
      const { asFragment } = render(
        <ForecastTrihourlySummary
          time={validProps.time}
          icon={validProps.icon}
          temp={validProps.temp}
          weather={validProps.weather}
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("props", () => {
    it("renders correct values for props", () => {
      const { getByText, getByTestId } = render(
        <ForecastTrihourlySummary
          time={validProps.time}
          icon={validProps.icon}
          temp={validProps.temp}
          weather={validProps.weather}
        />
      );

      expect(getByText(validProps.time)).toHaveAttribute(
        "class",
        "forecast-trihourly-summary__time"
      );
      expect(getByText(validProps.weather).closest("div")).toHaveAttribute(
        "class",
        "forecast-trihourly-summary__description"
      );
      expect(getByTestId("forecast-icon")).toHaveAttribute(
        "class",
        "forecast-trihourly-summary__icon"
      );
      expect(getByText(`${validProps.temp.toFixed(1)}K`)).toHaveAttribute(
        "class",
        "forecast-trihourly-summary__temp"
      );
    });
  });
});
