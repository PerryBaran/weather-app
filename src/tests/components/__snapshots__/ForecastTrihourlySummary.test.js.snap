// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastTrihourlySummary snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-trihourly-summary"
    data-testid="forecast-trihourly-summary"
  >
    <p>
      15:00
    </p>
    <div
      class="forecast-trihourly-summary__icon"
      data-testid="forecast-icon"
    >
      <i
        class="wi wi-day-storm-showers"
        name="owm"
      />
    </div>
    <p>
      <span>
        broken
      </span>
      <span
        class="forecast-trihourly-summary__line-break"
      >
        clouds
      </span>
    </p>
    <p>
      34.0Â°C
    </p>
  </div>
</DocumentFragment>
`;
